nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js

```diff
@@ -1272,11 +1301,9 @@
             }
 
             // initialize the connection dialog
-            $.when(initializeSourceNewConnectionDialog(source), initializeDestinationNewConnectionDialog(destination)).done(function () {
+            $.when(initializeSourceNewConnectionDialog(source), initializeDestinationNewConnectionDialog(destination), loadBackPressureDefaults()).done(function () {
```

Comment:
This approach is going to query for the backpressure defaults every time a new connection is opened. Since these values are not changing, we just need to load them once. If we end up using the existing /nifi-api/flow/config endpoint, we won't need to query for these here at all. Instead, the default values can be passed into the `nfConnectionConfiguration.init(...)` function. This function is invoked inside of the response handler of the flow configuration endpoint. Inside of the `init(...)` function we could just hold a reference to these values that could be restored every time the new connection dialog is opened.